torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -201,12 +203,13 @@ std::string wireSerialize(
     pickler.protocol();
     pickler.pushIValue(tensors);
     pickler.stop();
-    auto writeable_tensors = pickler.tensorData();
+    // tensorData is in function scope so that the data() pointers stay valid.
+    tensorData = pickler.tensorData();
```

Comment:
How expensive is it to capture by value here? I would assume `WriteableTensorData` is just a thin wrapper.  If not, shall we add some microbenchmark to see how much perf do we lose?